### ISSUE TITLE
Verification of `eth_getBlockNumber` for prover should not forward request

### ISSUE THREAD

**Issue by nazarhussain** [ChainSafe/lodestar#5796]
### Describe the bug

Currently we documented that `eth_blockNumber` will be forwarded to EL. This may cause to return the wrong block number which is not synced with the light client. 

### Expected behavior

Return the latest block number signed by the current sync-committee. 

### Operating system

Linux

### Lodestar version or commit hash

1.9.x

**Comment by mkermani144** [comment]
Is it sufficient/correct to create a new verified request for `eth_blockNumber` and use `getHead` of light client to fix it?

**Comment by nflaig** [comment]
> Is it sufficient/correct to create a new verified request for `eth_blockNumber` and use `getHead` of light client to fix it?

cc @nazarhussain

**Comment by nazarhussain** [comment]
> Is it sufficient/correct to create a new verified request for `eth_blockNumber` and use `getHead` of light client to fix it?

Yes this could be similar to existing implementation of `eth_getBlockByNumber` and instead of returning the block we can send the number of the block. 

If requested parameter is `finalized` or `latest` then we can get infor from our `PayloadStore` and get relevant block number for that hash. 

Looking forward to review it.

**Comment by TechFusionData** [comment]
Hi, I'd like to work on this issue. I'm a TypeScript developer looking to contribute to Lodestar and this looks like a good starting point. My understanding is that `eth_getBlockNumber` currently forwards the request to the execution layer, but for the prover it should instead return the latest block number that has been signed by the sync committee. I'll start by reviewing the prover package to understand the current request flow and then implement the fix. Happy to discuss approach before coding if that's preferred.
